src/simulators/matrix_product_state/matrix_product_state_tensor.hpp

```diff
@@ -459,30 +462,23 @@ void MPS_Tensor::contract_2_dimensions(const MPS_Tensor &left_gamma,
     throw std::runtime_error("left_size != right_size");
   result.resize(left_rows, right_columns);
 
-  #ifdef _WIN32
-     #pragma omp parallel for
-  #else
-     #pragma omp parallel for collapse(2)
-  #endif
-  for (int_t l_row=0; l_row<left_rows; l_row++)
-    for (int_t r_col=0; r_col<right_columns; r_col++)
-      result(l_row, r_col) = 0;
-  
-  #ifdef _WIN32
-     #pragma omp parallel for
-  #else
-     #pragma omp parallel for collapse(2)
-  #endif
-  for (int_t l_row=0; l_row<left_rows; l_row++)
-    for (int_t r_col=0; r_col<right_columns; r_col++) {
-
-      for (int_t size=0; size<left_size; size++)
-	  for (int_t index=0; index<left_columns ; index++) {
- 	      result(l_row, r_col) += left_gamma.data_[size](l_row, index) *
-		                      right_gamma.data_[size](index, r_col);      
-
-	    }
-	  }
+  uint_t omp_limit = left_rows*right_columns;
+#pragma omp parallel for collapse(2) if ((omp_limit > 10) && (omp_threads > 1)) num_threads(omp_threads) 
```

Comment:
So instead of using `omp_threads` it would be great if we can use: `MPS::get_num_threads()` and avoid passing `omp_threads` as a parameter.